infrastructure/repositories/event_repository.py

```diff
@@ -0,0 +1,24 @@
+from datetime import datetime as dt
+from sqlalchemy.exc import SQLAlchemyError
+from infrastructure.repositories.base_repository import BaseRepository
+from infrastructure.models import Event
+
+
+class EventRepository(BaseRepository):
+    def update_events(self, contract_id, contract_name, raw_events):
```

Comment:
define decorator for all repository functions please ?

Reply:
Will check on this. Will make this change when we bring in the domain object.